component/discovery/aws/lightsail.go

```diff
@@ -28,15 +28,14 @@ func init() {
 
 // LightsailArguments is the configuration for AWS Lightsail based service discovery.
 type LightsailArguments struct {
-	Endpoint        string            `river:"endpoint,attr,optional"`
-	Region          string            `river:"region,attr,optional"`
-	AccessKey       string            `river:"access_key,attr,optional"`
-	SecretKey       rivertypes.Secret `river:"secret_key,attr,optional"`
-	Profile         string            `river:"profile,attr,optional"`
-	RoleARN         string            `river:"role_arn,attr,optional"`
-	RefreshInterval time.Duration     `river:"refresh_interval,attr,optional"`
-	Port            int               `river:"port,attr,optional"`
-
+	Endpoint         string                  `river:"endpoint,attr,optional"`
+	Region           string                  `river:"region,attr,optional"`
+	AccessKey        string                  `river:"access_key,attr,optional"`
+	SecretKey        rivertypes.Secret       `river:"secret_key,attr,optional"`
+	Profile          string                  `river:"profile,attr,optional"`
```

Comment:
Sorry to leave another stray comment. Should some of these be non-optional since they don't have a default value set? https://prometheus.io/docs/prometheus/latest/configuration/configuration/#lightsail_sd_config

Reply:
I think they are truly optional. It is not possible to state what the default value will be, because the defaults come from environment variables.